python/rpdk/java/templates/init/guided_aws/TagHelper.java

```diff
@@ -0,0 +1,256 @@
+        final Map<String, String> tagMap = new HashMap<>();
+
+        if (handlerRequest.getSystemTags() != null) {
+            tagMap.putAll(handlerRequest.getSystemTags());
+        }
+
+        if (handlerRequest.getDesiredResourceTags() != null) {
+            tagMap.putAll(handlerRequest.getDesiredResourceTags());
+        }
+
+        // TODO: get tags from resource model based on your tag property name
+        // TODO: tagMap.putAll(convertToMap(resourceModel.getTags()));
+        return Collections.unmodifiableMap(tagMap);
+    }
+
+    /**
+     * shouldUpdateTags
+     *
+     * Determines whether user defined tags have been changed during update.
+     */
+    public final boolean shouldUpdateTags(final ResourceModel resourceModel, final ResourceHandlerRequest<ResourceModel> handlerRequest) {
+        final Map<String, String> currentTags = getPreviouslyAttachedTags(handlerRequest);
+        final Map<String, String> desiredTags = getNewDesiredTags(resourceModel, handlerRequest);
+        return ObjectUtils.notEqual(currentTags, desiredTags);
+    }
+
+    /**
+     * getPreviouslyAttachedTags
+     *
+     * If stack tags and resource tags are not merged together in Configuration class,
+     * we will get previous attached user defined tags from both handlerRequest.getPreviousResourceTags (stack tags)
+     * and handlerRequest.getPreviousResourceState (resource tags).
+     */
+    public Map<String, String> getPreviouslyAttachedTags(final ResourceHandlerRequest<ResourceModel> handlerRequest) {
+        // get previous stack level tags from handlerRequest
+        final Map<String, String> currentTags = handlerRequest.getPreviousResourceTags();
+
+        // TODO: get resource level tags from previous resource state based on your tag property name
+        // TODO: currentTags.putAll(handlerRequest.getPreviousResourceState().getTags());
+        return currentTags;
+    }
+
+    /**
+     * getNewDesiredTags
+     *
+     * If stack tags and resource tags are not merged together in Configuration class,
+     * we will get new user defined tags from both resource model and previous stack tags.
+     */
+    public Map<String, String> getNewDesiredTags(final ResourceModel resourceModel, final ResourceHandlerRequest<ResourceModel> handlerRequest) {
+        // get new stack level tags from handlerRequest
+        final Map<String, String> desiredTags = handlerRequest.getDesiredResourceTags();
+
+        // TODO: get tags from resource model based on your tag property name
+        // TODO: desiredTags.putAll(convertToMap(resourceModel.getTags()));
+        return desiredTags;
+    }
+
+    /**
+     * getTagsToAdd
+     *
+     * Determines the tags the customer desired to define or redefine.
+     */
+    public Map<String, String> generateTagsToAdd(final Map<String, String> currentTags, final Map<String, String> desiredTags) {
```

Comment:
Need to keep the name consistent. We should always call `previous` and `desired`, which means what is before and what is should be.  Don't use `current` term to make it confuse.

`currentTags` -> `previousTags`.

Reply:
nice catch, yeah will change it